src/gui/UnlockDatabaseDialog.h

```diff
@@ -0,0 +1,35 @@
+#ifndef KEEPASSX_AUTOTYPEUNLOCKDIALOG_H
+#define KEEPASSX_AUTOTYPEUNLOCKDIALOG_H
+
+#include <QAbstractItemModel>
+#include <QObject>
+#include <QDialog>
+
+#include <gui/DatabaseTabWidget.h>
+
+#include "core/Global.h"
+
+class UnlockDatabaseWidget;
+class Database;
+
+class UnlockDatabaseDialog : public QDialog
+{
+    Q_OBJECT
+public:
+    explicit UnlockDatabaseDialog(QWidget *parent = Q_NULLPTR);
+    void setDBFilename(const QString& filename);
+    void clearForms();
+    Database* database();
+
+Q_SIGNALS:
+    void unlockDone(bool);
+
+public Q_SLOTS:
+    void Done(bool r);
```

Comment:
Please rename as methods start with a lower case letter and done() already exists in QDialog.